### PR TITLE
chore: add support for staticcheck

### PR DESCRIPTION
**Pull request by remyleone**

### CODE REVIEW

File: internal/services/instance/ip_test.go [hunk 1]
```diff
@@ -308,8 +308,8 @@ func isIPAttachedToServer(tt *acctest.TestTools, ipResource, serverResource stri
 			return err
 		}
 
-		if server.Server.PublicIP.Address.String() != ip.IP.Address.String() {
-			return fmt.Errorf("IPs should be the same in %s and %s: %v is different than %v", ipResource, serverResource, server.Server.PublicIP.Address, ip.IP.Address)
+		if server.Server.PublicIP != nil && server.Server.PublicIP.Address.String() != ip.IP.Address.String() { //nolint:staticcheck
```
Comment (remyleone) [inline internal/services/instance/ip_test.go:311]: We need to add test and documentation to ensure that we detect when this field is removed from API responses and that the user are warned that the field is deprecated.

File: internal/services/instance/servers_data_source.go [hunk 2]
```diff
@@ -165,8 +165,8 @@ func DataSourceInstanceServersRead(ctx context.Context, d *schema.ResourceData,
 	for _, server := range res.Servers {
 		rawServer := make(map[string]interface{})
 		rawServer["id"] = zonal.NewID(server.Zone, server.ID).String()
-		if server.PublicIP != nil {
-			rawServer["public_ip"] = server.PublicIP.Address.String()
+		if server.PublicIP != nil { //nolint:staticcheck
```
Comment (remyleone) [inline internal/services/instance/servers_data_source.go:168]: We need to add test and documentation to ensure that we detect when this field is removed from API responses and that the user are warned that the field is deprecated.

File: internal/services/instance/servers_data_source.go [hunk 3]
```diff
@@ -183,15 +183,17 @@ func DataSourceInstanceServersRead(ctx context.Context, d *schema.ResourceData,
 		rawServer["zone"] = string(zone)
 		rawServer["name"] = server.Name
 		rawServer["boot_type"] = server.BootType
-		if server.Bootscript != nil {
-			rawServer["bootscript_id"] = server.Bootscript.ID
+		if server.Bootscript != nil { //nolint:staticcheck
```
Comment (remyleone) [inline internal/services/instance/servers_data_source.go:186]: We need to add test and documentation to ensure that we detect when this field is removed from API responses and that the user are warned that the field is deprecated.

File: internal/services/instance/servers_data_source.go [hunk 4]
```diff
@@ -183,15 +183,17 @@ func DataSourceInstanceServersRead(ctx context.Context, d *schema.ResourceData,
...
+			rawServer["bootscript_id"] = server.Bootscript.ID //nolint:staticcheck
 		}
 		rawServer["type"] = server.CommercialType
 		if len(server.Tags) > 0 {
 			rawServer["tags"] = server.Tags
 		}
 		rawServer["security_group_id"] = zonal.NewID(zone, server.SecurityGroup.ID).String()
-		rawServer["enable_ipv6"] = server.EnableIPv6
+		if server.EnableIPv6 != nil { //nolint:staticcheck
```
Comment (remyleone) [inline internal/services/instance/servers_data_source.go:194]: We need to add test and documentation to ensure that we detect when this field is removed from API responses and that the user are warned that the field is deprecated.

File: internal/services/instance/servers_data_source.go [hunk 5]
```diff
@@ -203,10 +205,10 @@ func DataSourceInstanceServersRead(ctx context.Context, d *schema.ResourceData,
 			rawServer["placement_group_id"] = zonal.NewID(zone, server.PlacementGroup.ID).String()
 			rawServer["placement_group_policy_respected"] = server.PlacementGroup.PolicyRespected
 		}
-		if server.IPv6 != nil {
-			rawServer["ipv6_address"] = server.IPv6.Address.String()
-			rawServer["ipv6_gateway"] = server.IPv6.Gateway.String()
-			prefixLength, err := strconv.Atoi(server.IPv6.Netmask)
+		if server.IPv6 != nil { //nolint:staticcheck
```
Comment (remyleone) [inline internal/services/instance/servers_data_source.go:208]: We need to add test and documentation to ensure that we detect when this field is removed from API responses and that the user are warned that the field is deprecated.

File: internal/services/instance/types.go [hunk 6]
```diff
@@ -66,7 +66,7 @@ func flattenImageExtraVolumes(volumes map[string]*instance.Volume, zone scw.Zone
 		volumeFlat := map[string]interface{}{
 			"id":                zonal.NewIDString(zone, volume.ID),
 			"name":              volume.Name,
-			"export_uri":        volume.ExportURI,
+			"export_uri":        volume.ExportURI, //nolint:staticcheck
```
Comment (remyleone) [inline internal/services/instance/types.go:69]: We need to add test and documentation to ensure that we detect when this field is removed from API responses and that the user are warned that the field is deprecated.

File: internal/services/lb/frontend.go [hunk 7]
```diff
@@ -316,8 +316,8 @@ func resourceLbFrontendRead(ctx context.Context, d *schema.ResourceData, m inter
 	_ = d.Set("timeout_client", types.FlattenDuration(frontend.TimeoutClient))
 	_ = d.Set("enable_http3", frontend.EnableHTTP3)
 
-	if frontend.Certificate != nil {
-		_ = d.Set("certificate_id", zonal.NewIDString(zone, frontend.Certificate.ID))
+	if frontend.Certificate != nil { //nolint:staticcheck
```
Comment (remyleone) [inline internal/services/lb/frontend.go:319]: We need to add test and documentation to ensure that we detect when this field is removed from API responses and that the user are warned that the field is deprecated.

File: internal/services/lb/helpers_lb.go [hunk 8]
```diff
@@ -56,10 +56,10 @@ func IsPrivateNetworkEqual(a, b *lbSDK.PrivateNetwork) bool {
 	if a.PrivateNetworkID != b.PrivateNetworkID {
 		return false
 	}
-	if !reflect.DeepEqual(a.DHCPConfig, b.DHCPConfig) {
+	if !reflect.DeepEqual(a.DHCPConfig, b.DHCPConfig) { //nolint:staticcheck
```
Comment (remyleone) [inline internal/services/lb/helpers_lb.go:59]: We need to add test and documentation to ensure that we detect when this field is removed from API responses and that the user are warned that the field is deprecated.

File: internal/services/lb/helpers_lb.go [hunk 9]
```diff
@@ -56,10 +56,10 @@ func IsPrivateNetworkEqual(a, b *lbSDK.PrivateNetwork) bool {
...
 		return false
 	}
-	if !reflect.DeepEqual(a.StaticConfig, b.StaticConfig) {
+	if !reflect.DeepEqual(a.StaticConfig, b.StaticConfig) { //nolint:staticcheck
```
Comment (remyleone) [inline internal/services/lb/helpers_lb.go:62]: We need to add test and documentation to ensure that we detect when this field is removed from API responses and that the user are warned that the field is deprecated.

File: internal/services/tem/domain.go [hunk 10]
```diff
@@ -226,7 +226,7 @@ func ResourceDomainRead(ctx context.Context, d *schema.ResourceData, m interface
 	_ = d.Set("next_check_at", types.FlattenTime(domain.NextCheckAt))
 	_ = d.Set("last_valid_at", types.FlattenTime(domain.LastValidAt))
 	_ = d.Set("revoked_at", types.FlattenTime(domain.RevokedAt))
-	_ = d.Set("last_error", domain.LastError)
+	_ = d.Set("last_error", domain.LastError) //nolint:staticcheck
```
Comment (remyleone) [inline internal/services/tem/domain.go:229]: We need to add test and documentation to ensure that we detect when this field is removed from API responses and that the user are warned that the field is deprecated.

**Review by Codelax — APPROVED** [review]

**Pull request merged**